src/App.tsx

```diff
@@ -15,48 +27,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+function getPreparedGoods(goods: string[],
+  { sortBy, isReverse }: FilterParams) {
+  const sortedGoods = [...goods];
+
+  if (sortBy) {
+    sortedGoods.sort((good1, good2) => {
+      switch (sortBy) {
+        case SortType.SORT_BY_ABC:
+          return good1.localeCompare(good2);
+
+        case SortType.SORT_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverse) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+}
+
+function setReverse(isReverse : boolean,
+  setIsRevers: React.Dispatch<React.SetStateAction<boolean>>) {
+  return isReverse === false ? setIsRevers(true) : setIsRevers(false);
+}
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState(SortType.DEFAULT);
+  const [isReverse, setIsReverse] = useState(false);
+  const goods = getPreparedGoods(goodsFromServer, { sortBy, isReverse });
+
+  const resetSorting = () => {
+    setIsReverse(false);
+    setSortBy(SortType.DEFAULT);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button', 'is-info',
+            { 'is-light': sortBy !== SortType.SORT_BY_ABC })}
+          onClick={() => setSortBy(SortType.SORT_BY_ABC)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button', 'is-success',
+            { 'is-light': sortBy !== SortType.SORT_BY_LENGTH })}
+          onClick={() => setSortBy(SortType.SORT_BY_LENGTH)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button', 'is-warning',
+            { 'is-light': isReverse === false })}
```

Comment:
Instead of comparing 'isReverse' to false, you could use '!isReverse'. This makes the code cleaner and easier to understand.